lib/hooks/index.js

```diff
@@ -97,25 +58,45 @@ function findModuleVersion(request, parent, load) {
     if (mainScriptDir.indexOf(paths[i]) === 0) {
       // This is unsafe on windows if request contains a namespace
       // TODO: convert path separators in request
-      var pjson = load(path.join(paths[i], request, 'package.json'));
-      return pjson.version;
+      return path.join(paths[i], request);
     }
   }
   return null;
 }
 
+function findModuleVersion(modulePath, load) {
+  return modulePath ? load(path.join(modulePath, 'package.json')).version :
+    process.version;
+}
+
 function activate(agent) {
 
   logger = agent.logger;
 
-  function loadHookAndPatch(loadFn, instrumentation) {
-    assert(instrumentation.module);
-    assert(!instrumentation.hook);
-
+  function loadHookAndPatch(loadFn, instrumentation, moduleRoot, version) {
```

Comment:
This caching does not appear to happen the way I expected. I would be inclined to open this as a bug and fix it independently as this functionality wasn't present before this change and it doesn't represent a regression introduced by this change. I expect this to be a relatively rare case since web servers and databases are rarely buried in dependencies the way utilities are.